execution-engine/contracts/hdac-system/pop-install/src/lib.rs

```diff
@@ -63,25 +68,14 @@ pub extern "C" fn call() {
         .map(|key| (key, PLACEHOLDER_KEY))
         .collect();
 
-    // Insert genesis validator's delegations.
-    // We also store delegations in the form key:
-    // "d_{delegator_pk}_{validator_pk}_{delegation_amount}", value: doesn't matter
-    genesis_validators
-        .iter()
-        .map(|(pub_key, balance)| {
-            let key_bytes = pub_key.value();
-            let mut hex_key = String::with_capacity(64);
-            for byte in &key_bytes[..32] {
-                write!(hex_key, "{:02x}", byte).unwrap();
-            }
-            let mut uref = String::new();
-            uref.write_fmt(format_args!("d_{}_{}_{}", hex_key, hex_key, balance))
-                .unwrap();
-            uref
-        })
-        .for_each(|key| {
-            named_keys.insert(key, PLACEHOLDER_KEY);
-        });
+    // Insert genesis state information.
+    // We also store in the form key:
+    let state_informations: Vec<String> = runtime::get_arg(Args::StateInformations as u32)
+        .unwrap_or_revert_with(ApiError::MissingArgument)
+        .unwrap_or_revert_with(ApiError::InvalidArgument);
+    state_informations.iter().for_each(|key| {
+        named_keys.insert(key.to_string(), PLACEHOLDER_KEY);
+    });
```

Comment:
I hope we have basic validation logic here. This version totally relies on the assumption that a user will pass the perfectly formed input. For example, if the user misses a self-delegation by mistake, this will lead to a system error.

Reply:
Complete to change.
https://github.com/hdac-io/CasperLabs/pull/22/commits/982815a89be3e065345db3f2544e927b57eac272